RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -0,0 +1,965 @@
+      Var_woGwGSF_Endcap_[17] = std::log(std::max(float(1.), TauPt));
+      Var_woGwGSF_Endcap_[18] = TauEmFraction;
+      Var_woGwGSF_Endcap_[19] = TauLeadPFChargedHadrHoP;
+      Var_woGwGSF_Endcap_[20] = TauLeadPFChargedHadrEoP;
+      Var_woGwGSF_Endcap_[21] = TauVisMassIn;
+      Var_woGwGSF_Endcap_[22] = TaudCrackEta;
+      mvaValue = mva_woGwGSF_EC_->GetClassifier(Var_woGwGSF_Endcap_);
+    } 
+  }
+  else if ( TauSignalPFGammaCandsIn > 0 && TauHasGsf > 0.5 ) {
+    if ( std::abs(TauEtaAtEcalEntrance) < 1.479 ) {
+      Var_wGwGSF_Barrel_[0]  = std::max(float(-0.1), ElecEtotOverPin);
+      Var_wGwGSF_Barrel_[1]  = std::log(ElecChi2NormGSF);
+      Var_wGwGSF_Barrel_[2]  = ElecGSFNumHits;
+      Var_wGwGSF_Barrel_[3]  = std::log(ElecGSFTrackResol);
+      Var_wGwGSF_Barrel_[4]  = ElecGSFTracklnPt;
+      Var_wGwGSF_Barrel_[5]  = ((ElecGSFNumHits - ElecKFNumHits)/(ElecGSFNumHits + ElecKFNumHits));
+      Var_wGwGSF_Barrel_[6]  = std::log(ElecChi2NormKF);
+      Var_wGwGSF_Barrel_[7]  = std::min(std::abs(ElecPin - ElecPout)/ElecPin, float(1.));
+      Var_wGwGSF_Barrel_[8]  = std::min(ElecEecal/ElecPout, float(20.));
+      Var_wGwGSF_Barrel_[9]  = ElecDeltaEta;
+      Var_wGwGSF_Barrel_[10] = ElecDeltaPhi;
+      Var_wGwGSF_Barrel_[11] = std::min(ElecMvaInSigmaEtaEta, float(0.01));
+      Var_wGwGSF_Barrel_[12] = std::min(ElecMvaInHadEnergy, float(20.));
+      Var_wGwGSF_Barrel_[13] = std::min(ElecMvaInDeltaEta, float(0.1));
+      Var_wGwGSF_Barrel_[14] = TauEtaAtEcalEntrance;
+      Var_wGwGSF_Barrel_[15] = TauLeadChargedPFCandEtaAtEcalEntrance;
+      Var_wGwGSF_Barrel_[16] = std::min(float(2.), TauLeadChargedPFCandPt/std::max(float(1.), TauPt));
+      Var_wGwGSF_Barrel_[17] = std::log(std::max(float(1.), TauPt));
+      Var_wGwGSF_Barrel_[18] = TauEmFraction;
+      Var_wGwGSF_Barrel_[19] = TauSignalPFGammaCandsIn;
+      Var_wGwGSF_Barrel_[20] = TauSignalPFGammaCandsOut;
+      Var_wGwGSF_Barrel_[21] = TauLeadPFChargedHadrHoP;
+      Var_wGwGSF_Barrel_[22] = TauLeadPFChargedHadrEoP;
+      Var_wGwGSF_Barrel_[23] = TauVisMassIn;
+      Var_wGwGSF_Barrel_[24] = TauGammaEtaMomIn;
+      Var_wGwGSF_Barrel_[25] = TauGammaEtaMomOut;
+      Var_wGwGSF_Barrel_[26] = TauGammaPhiMomIn;
+      Var_wGwGSF_Barrel_[27] = TauGammaPhiMomOut;
+      Var_wGwGSF_Barrel_[28] = TauGammaEnFracIn;
+      Var_wGwGSF_Barrel_[29] = TauGammaEnFracOut;
+      Var_wGwGSF_Barrel_[30] = TaudCrackEta;
+      Var_wGwGSF_Barrel_[31] = TaudCrackPhi;
+      mvaValue = mva_wGwGSF_BL_->GetClassifier(Var_wGwGSF_Barrel_);
+    } else {
+      Var_wGwGSF_Endcap_[0]  = std::max(float(-0.1), ElecEtotOverPin);
+      Var_wGwGSF_Endcap_[1]  = std::log(ElecChi2NormGSF);
+      Var_wGwGSF_Endcap_[2]  = ElecGSFNumHits;
+      Var_wGwGSF_Endcap_[3]  = std::log(ElecGSFTrackResol);
+      Var_wGwGSF_Endcap_[4]  = ElecGSFTracklnPt;
+      Var_wGwGSF_Endcap_[5]  = ((ElecGSFNumHits - ElecKFNumHits)/(ElecGSFNumHits + ElecKFNumHits));
```

Comment:
Need more checks against division by 0: here and lines 399, 422, and 450.
Also, please replace magic number 1.479 with constant throughout.